src/ophyd_async/epics/areadetector/drivers/ad_base.py

```diff
@@ -52,6 +54,36 @@ def __init__(self, prefix: str, name: str = "") -> None:
         super().__init__(prefix, name=name)
 
 
+async def set_exposure_time_and_acquire_period_if_supplied(
+    controller: DetectorControl,
```

Comment:
We pass the controller here to get the detector's deadtime, which we can then add to the exposure time to make an acquire period. This works but may be misleading as a longer deadtime could be passed into the detector via `TriggerInfo`. It is unclear whether that deadtime should be followed here, it will never be smaller than the result of `get_deadtime` and the `TriggerInfo` is not available from the controller, so some refactoring and potential API changes would be required.

Obeying the output of `get_deadtime` gives us the following if `TriggerInfo` supplies a larger value:
![image](https://github.com/bluesky/ophyd-async/assets/29771545/590ddc31-9f05-4824-a53b-32ae47b1c859)

Which is quite nice as it gives us the freedom to configure the trriger system to delay the frame to the middle of the duty cycle, which is useful in preventing things like snaking:
![image](https://github.com/bluesky/ophyd-async/assets/29771545/99e5b473-acaf-41e5-a0ed-32aaa72abc86)

Using the `TriggerInfo` deadtime seems more restrictive but I believe that is how Malcolm works:
![image](https://github.com/bluesky/ophyd-async/assets/29771545/106a667f-c6b5-4755-a2a4-1d72dad2518e)

Reply:
Let's use `TriggerInfo.deadtime`. This means we need to consider the interface between `StandardDetector` and `DetectorController`. Opened https://github.com/bluesky/ophyd-async/issues/405 to track

Reply:
The middle diagram does concern me in that case, how do we prevent snaking artefacts in grid scans?

Reply:
I believe I have now clarified this in #405